### PR TITLE
MAR-33: Schema for tracking provider assignment

### PR DESCRIPTION
**Pull request by evan-duncan**

### CODE REVIEW

**Comment by alexsmithmn** [comment]
Should the work order ID get pulled in here or does that happen elsewhere?

File: schemas/com.fieldnation/assign_provider_action/jsonschema/1-0-0 [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            "items": {
+                "type": "number"
+            }
+        },
+        "view": {
+            "enum": ["list", "map"]
+        }
+	},
+	"required": ["provider_id", "available_providers", "presented_providers", "requested_providers", "declined_providers", "view"],
+	"additionalProperties": false
```
Comment (amran-mamoon-fn) [inline schemas/com.fieldnation/assign_provider_action/jsonschema/1-0-0]: Bit confused here. Don't we need to pass additional context here like workOrder, userInterface etc.?

Comment (evan-duncan) [inline schemas/com.fieldnation/assign_provider_action/jsonschema/1-0-0]: This is for additional properties on just this model. When you actually track an event there is an arbitrary number of contexts that you can include. This type of event tracking ends up looking like

```js
snowplow('trackSelfDescribingEvent', {
    schema: 'assign_provider_action',
    data: { ... }
}, contexts);
```

**Comment by evan-duncan** [comment]
@alexsmithmn @amran-mamoon-fn I had to change up this schema a little bit because of the limitations of the API. It's pretty much impossible for accurately track what providers are shown on tabs that aren't being shown. In this case our primary objective is to figure out what provider is being assigned based on the options available in the requested list. I've changed this schema to be a little more generic so we can start tracking other actions on the find provider screen. This is what a sample request body will look like.

```js
{
  type: 'SNOWPLOW_UNSTRUCTURED_EVENT',
  schema: 'iglu:com.fieldnation/find_provider_action/jsonschema/1-0-0',
  data: {
    provider_list: 'requested',
    provider_id: 558,
    providers: [
      557,
      558,
      559
    ],
    provider_position: 1,
    view: 'list',
    current_page: 1,
    providers_per_page: 25
  },
  contexts: [
    {
      schema: 'iglu:com.fieldnation/work_order_context/jsonschema/1-0-0',
      data: {
        work_order_id: '61'
      }
    },
    {
      schema: 'iglu:com.fieldnation/user_interface_context/jsonschema/1-0-0',
      data: {
        'interface': 'React',
        version: '12.0.5-2-gb99bbc0',
        page: '#61 Alarm system false reporting due to possible keypad water damage :: Field Nation'
      }
    }
  ]
}
```

**Comment by amran-mamoon-fn** [comment]
Make sense.

**Pull request merged**